test/test_torch.py

```diff
@@ -5221,6 +5221,15 @@ def test_multiplication_numpy_scalar(self):
                 self.assertTrue(r2.dtype == t_dtype)
                 self.assertTrue(r2.requires_grad)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_parse_numpy_int(self):
+        # https://github.com/pytorch/pytorch/issues/29252
+        np_val = np.ones(1, dtype=np.int64)[0]
+        self.assertEqual(type(np_val), np.int64)
+        th_val = torch.ones(size=(1,), dtype=torch.long)
```

Comment:
nit: If you just wanted a scalar you can do `torch.ones([], dtype=torch.long)`

Reply:
I do want 1 dim not zero-dim though here

Reply:
Why?

Reply:
You said "If you just wanted a scalar," but this isn't a scalar. I'm not exactly sure what you mean.

The example came from the specific reported bug.  I wrote the test first based on the given example, verified that the test failed, then made the changes to make the test pass.

Reply:
> The example came from the specific reported bug. I wrote the test first based on the given example, verified that the test failed, then made the changes to make the test pass.

Thanks, that's the explanation I was looking for